frame/beefy/src/lib.rs

```diff
@@ -146,19 +159,25 @@ impl<T: Config> OneSessionHandler<T::AccountId> for Pallet<T> {
 		I: Iterator<Item = (&'a T::AccountId, T::BeefyId)>,
 	{
 		let authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
-		Self::initialize_authorities(&authorities);
+		// we panic here as runtime maintainers can simply reconfigure genesis and restart the
+		// chain easily
+		Self::initialize_authorities(&authorities).expect("Authorities vec too big");
 	}
 
 	fn on_new_session<'a, I: 'a>(_changed: bool, validators: I, queued_validators: I)
 	where
 		I: Iterator<Item = (&'a T::AccountId, T::BeefyId)>,
 	{
 		let next_authorities = validators.map(|(_, k)| k).collect::<Vec<_>>();
+		let bounded_next_authorities =
+			BoundedVec::<_, T::MaxAuthorities>::truncate_from(next_authorities);
 		let next_queued_authorities = queued_validators.map(|(_, k)| k).collect::<Vec<_>>();
+		let bounded_next_queued_authorities =
+			BoundedVec::<_, T::MaxAuthorities>::truncate_from(next_queued_authorities);
```

Comment:
Does this tell us if the value was truncated?

Ideally we would print a simple log message about that behavior.

Reply:
No, and you can't really make a log unfortunately, unless I specifically do a length check before calling `truncate_from`.

Reply:
Let's do a length check, it's really cheap for a `Vec::Iterator` and this function is only called once per session.

Having list of validators truncated would almost always break BEEFY consensus so we definitely want at least an
```rust
log::error!(target: "runtime::beefy", "authorities list {} truncated to length {}.", next_queued_authorities, T::MaxAuthorities);
```
here.

Reply:
Let's do a length check, it's really cheap for a `Vec::Iterator` and this function is only called once per session.

Having list of validators truncated would almost always break BEEFY consensus so we definitely want at least an
```rust
log::error!(target: "runtime::beefy", "authorities list {} truncated to length {}.", next_queued_authorities, T::MaxAuthorities);
```
here.

Reply:
Done, but I had to use `{:?}` for the authorities list because `BeefyId` doesn't implement `Display`.